README.markdown

```diff
@@ -217,52 +218,29 @@ The example above demonstrates the following style guidelines:
 
 ### Use of Self
 
-For conciseness, avoid using `self` since Swift does not require it to access an object's properties or invoke its methods.
-
-Use `self` when required to differentiate between property names and arguments in initializers, and when referencing properties in closure expressions (as required by the compiler):
+Always use `self` when possible. Although Xcode's syntax highlighting makes distinguishing easier, the use of `self` makes it even easier. The distinguising comes in with local variables vs properties, and global functions vs instance/class functions/methods:
```

Comment:
Yeah, I don't vote for that either. 

The fact that Swift compiler ONLY forces self in the closure makes it easier to find the potential circular reference to self.